### ISSUE TITLE
Breaking API change(s) in xgboost4j 0.81

### ISSUE THREAD

**Issue by falaki** [combust/mleap#448]
XGBoost4j v0.81 includes this commit https://github.com/dmlc/xgboost/pull/3602 which changes the return type of `BoosterParams.getTreeLimit()` from `double` to `int`. This breaks model serialization because of [this line](https://github.com/combust/mleap/blob/master/mleap-xgboost-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/mleap/XGBoostClassificationModelOp.scala#L40:)
```
model.withValue("thresholds", thresholds.map(_.toSeq).map(Value.doubleList)).
    withValue("num_classes", Value.int(obj.numClasses)).
    withValue("num_features", Value.int(numFeatures)).
    withValue("tree_limit", Value.int(obj.getTreeLimit.toInt))
```

* I have not tested all other interactions with XGBoost models.
* The bigger question is how should we handle such breaking API changes from XGBoost. They seem to be fairly common. I can imagine two possibilities:
  * Release a new runtime for each new release of XGBoost
  * Handle breaking changes inside mleap (through shims or careful error handling).

**Comment by cchung100m** [comment]
Hi falaki,
PR #447 already identified the change of XGBoost 0.81 and the enhancement will be merged soon.

**Comment by falaki** [comment]
Thanks @calinrc. Have you tested #447 with xgboost v0.80? It seems to me we will fail that version. Is that patch going to be merged into version 0.13?

**Comment by ancasarb** [comment]
Closing this as PR#447 has been merged.
